### ISSUE TITLE
 enhancement: app doesnot work with android 2.2.2

### ISSUE THREAD

**Issue by arm2arm** [owncloud/android#889]
My old phone with 2.2.2 always gives Failure: "the application
ownCloud(process com.owncloud.android) has stopped unexpectedly.
Please try again." Force close or Report. i sent the report to google play.

I believe it is some marketing strategy of Android,
It is pity that 1Ghz phone with 15GB ram is unable to run OC client, I am running OC8 on rPi B (800Mhz+512MB).

If some one can look if it is easy to fix, it will be great to have a patch for an older Android.

**Comment by masensio** [comment]
Hi @arm2arm, 

I'm sorry. Unfortunately, we are not supporting Android 2.x anymore. 
But, in case of any volunteer wants to deal with it , it will be necessary more information about the problem because you give a generic error.
Unless any volunteer wants to deal with it, it won't be fixed.

**Comment by flyingOwl** [comment]
I'm on Android 2.3.7 (CyanogenMod 7.2) and also suffer from this crash. (Running v1.7.0 from F-Droid, which btw says Android 2.2 is supported)
The android logcat offers following information:

```
D/ViewConsistency(11712): AbsListView com.owncloud.android.ui.ExtendedListView@4055e870 enabled= true
D/ViewConsistency(11712): AbsListView com.owncloud.android.ui.ExtendedListView@4055e870 enabled= true
I/dalvikvm(11712): Could not find method android.widget.GridView.getColumnWidth, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.getColumnWidthCompatible
W/dalvikvm(11712): VFY: unable to resolve virtual method 8637: Landroid/widget/GridView;.getColumnWidth ()I
D/dalvikvm(11712): VFY: replacing opcode 0x6f at 0x0006
D/dalvikvm(11712): VFY: dead code 0x0009-0009 in Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.getColumnWidthCompatible ()I
I/dalvikvm(11712): Could not find method android.widget.GridView.getNumColumns, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.getNumColumnsCompatible
W/dalvikvm(11712): VFY: unable to resolve virtual method 8638: Landroid/widget/GridView;.getNumColumns ()I
D/dalvikvm(11712): VFY: replacing opcode 0x6f at 0x0006
D/dalvikvm(11712): VFY: dead code 0x0009-0009 in Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.getNumColumnsCompatible ()I
I/dalvikvm(11712): Could not find method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.smoothScrollToPositionFromTop, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.tryToScrollToBottomSmoothly
W/dalvikvm(11712): VFY: unable to resolve virtual method 20600: Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.smoothScrollToPositionFromTop (II)V
D/dalvikvm(11712): VFY: replacing opcode 0x6e at 0x0011
I/dalvikvm(11712): Could not find method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.smoothScrollToPositionFromTop, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.tryToScrollToBottomSmoothly
W/dalvikvm(11712): VFY: unable to resolve virtual method 20601: Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.smoothScrollToPositionFromTop (III)V
D/dalvikvm(11712): VFY: replacing opcode 0x6e at 0x0011
D/ViewConsistency(11712): AbsListView third_parties.in.srain.cube.GridViewWithHeaderAndFooter@4057bb28 enabled= true
D/ViewConsistency(11712): AbsListView third_parties.in.srain.cube.GridViewWithHeaderAndFooter@4057bb28 enabled= true
E/OCFileListFragment(11712): onActivityCreated() start
I/dalvikvm(11712): Could not find method android.widget.AbsListView.getChoiceMode, referenced from method com.owncloud.android.ui.adapter.FileListListAdapter.getView
W/dalvikvm(11712): VFY: unable to resolve virtual method 8518: Landroid/widget/AbsListView;.getChoiceMode ()I
D/dalvikvm(11712): VFY: replacing opcode 0x74 at 0x0152
D/dalvikvm(11712): VFY: dead code 0x0155-015e in Lcom/owncloud/android/ui/adapter/FileListListAdapter;.getView (ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
D/dalvikvm(11712): VFY: dead code 0x0227-024a in Lcom/owncloud/android/ui/adapter/FileListListAdapter;.getView (ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
D/AndroidRuntime(11712): Shutting down VM
W/dalvikvm(11712): threadid=1: thread exiting with uncaught exception (group=0x40015560)
E/AndroidRuntime(11712): FATAL EXCEPTION: main
E/AndroidRuntime(11712): java.lang.NoSuchMethodError: android.widget.AbsListView.setAdapter
E/AndroidRuntime(11712): 	at com.owncloud.android.ui.fragment.ExtendedListFragment.setListAdapter(ExtendedListFragment.java:84)
E/AndroidRuntime(11712): 	at com.owncloud.android.ui.fragment.OCFileListFragment.onActivityCreated(OCFileListFragment.java:137)
E/AndroidRuntime(11712): 	at android.support.v4.app.Fragment.performActivityCreated(Fragment.java:1794)
E/AndroidRuntime(11712): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:967)
E/AndroidRuntime(11712): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1126)
E/AndroidRuntime(11712): 	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:739)
E/AndroidRuntime(11712): 	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1489)
E/AndroidRuntime(11712): 	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:548)
E/AndroidRuntime(11712): 	at com.owncloud.android.ui.activity.FileActivity.onStart(FileActivity.java:182)
E/AndroidRuntime(11712): 	at com.owncloud.android.ui.activity.FileDisplayActivity.onStart(FileDisplayActivity.java:220)
E/AndroidRuntime(11712): 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1129)
E/AndroidRuntime(11712): 	at android.app.Activity.performStart(Activity.java:3791)
E/AndroidRuntime(11712): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1731)
E/AndroidRuntime(11712): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1784)
E/AndroidRuntime(11712): 	at android.app.ActivityThread.access$1500(ActivityThread.java:123)
E/AndroidRuntime(11712): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:939)
E/AndroidRuntime(11712): 	at android.os.Handler.dispatchMessage(Handler.java:99)
E/AndroidRuntime(11712): 	at android.os.Looper.loop(Looper.java:130)
E/AndroidRuntime(11712): 	at android.app.ActivityThread.main(ActivityThread.java:3835)
E/AndroidRuntime(11712): 	at java.lang.reflect.Method.invokeNative(Native Method)
E/AndroidRuntime(11712): 	at java.lang.reflect.Method.invoke(Method.java:507)
E/AndroidRuntime(11712): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:864)
E/AndroidRuntime(11712): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:622)
E/AndroidRuntime(11712): 	at dalvik.system.NativeStart.main(Native Method)
W/ActivityManager( 2394):   Force finishing activity com.owncloud.android/.authentication.AuthenticatorActivity
W/ActivityManager( 2394):   Force finishing activity com.owncloud.android/.ui.activity.FileDisplayActivity
W/ActivityManager( 2394): Activity pause timeout for HistoryRecord{40d47bd8 com.owncloud.android/.authentication.AuthenticatorActivity}
D/dalvikvm( 2394): GC_CONCURRENT freed 1351K, 59% free 7841K/18951K, external 806K/1304K, paused 8ms+17ms
```

**Comment by jab416171** [comment]
@masensio if owncloud doesn't support 2.x, the minimum API in the manifest should be updated to reflect that (at least until a volunteer updates the app so it supports 2.x again)

**Comment by TtuxX** [comment]
So sad owncloud client stopped working on android 2.x :'(
As flyingOwl, I am on Android 2.3.7 (MIUI 2.3.9) and also experiment this systematic crash. (*Also* running v1.7.0 from F-Droid, which *also* says Android 2.2 is supported).

Let's hope someone will make owncloud android client supports 2.x again, that would really be awesome
 
Crashlog app returns the following error messages:

~~~
placing opcode 0x71 at 0x0043
02-22 07:49:04.738 D/dalvikvm( 9159): VFY: dead code 0x0046-004e in Lcom/actionbarsherlock/internal/widget/IcsProgressBar;.onMeasure (II)V
02-22 07:49:04.788 I/dalvikvm( 9159): Could not find method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.setChoiceMode, referenced from method com.owncloud.android.ui.fragment.ExtendedListFragment.setChoiceMode
02-22 07:49:04.788 W/dalvikvm( 9159): VFY: unable to resolve virtual method 20591: Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.setChoiceMode (I)V
02-22 07:49:04.788 D/dalvikvm( 9159): VFY: replacing opcode 0x6e at 0x0007
02-22 07:49:04.788 D/dalvikvm( 9159): VFY: dead code 0x000a-000a in Lcom/owncloud/android/ui/fragment/ExtendedListFragment;.setChoiceMode (I)V
02-22 07:49:04.788 I/dalvikvm( 9159): Could not find method android.widget.AbsListView.setAdapter, referenced from method com.owncloud.android.ui.fragment.ExtendedListFragment.setListAdapter
02-22 07:49:04.788 W/dalvikvm( 9159): VFY: unable to resolve virtual method 8525: Landroid/widget/AbsListView;.setAdapter (Landroid/widget/ListAdapter;)V
02-22 07:49:04.788 D/dalvikvm( 9159): VFY: replacing opcode 0x6e at 0x0004
02-22 07:49:04.788 D/dalvikvm( 9159): VFY: dead code 0x0007-000c in Lcom/owncloud/android/ui/fragment/ExtendedListFragment;.setListAdapter (Landroid/widget/ListAdapter;)V
02-22 07:49:04.803 E/FileDisplayActivity( 9159): OCFileListFragment is null
02-22 07:49:04.803 D/FileDisplayActivity( 9159): onCreate() end
02-22 07:49:04.803 E/OCFileListFragment( 9159): onAttach
02-22 07:49:04.803 D/ExtendedListFragment( 9159): onCreateView
02-22 07:49:04.858 W/dalvikvm( 9159): PGC: map but no entry for Landroid/view/LayoutInflater;.inflate addr=0xfff8e76a
02-22 07:49:04.953 D/dalvikvm( 9159): GC_CONCURRENT freed 424K, 20% free 7525K/9351K, external 528K/532K, paused 9ms+4ms
02-22 07:49:04.953 I/dalvikvm( 9159): Could not find method android.widget.GridView.getColumnWidth, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.getColumnWidthCompatible
02-22 07:49:04.953 W/dalvikvm( 9159): VFY: unable to resolve virtual method 8637: Landroid/widget/GridView;.getColumnWidth ()I
02-22 07:49:04.953 D/dalvikvm( 9159): VFY: replacing opcode 0x6f at 0x0006
02-22 07:49:04.953 D/dalvikvm( 9159): VFY: dead code 0x0009-0009 in Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.getColumnWidthCompatible ()I
02-22 07:49:04.958 I/dalvikvm( 9159): Could not find method android.widget.GridView.getNumColumns, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.getNumColumnsCompatible
02-22 07:49:04.958 W/dalvikvm( 9159): VFY: unable to resolve virtual method 8638: Landroid/widget/GridView;.getNumColumns ()I
02-22 07:49:04.958 D/dalvikvm( 9159): VFY: replacing opcode 0x6f at 0x0006
02-22 07:49:04.958 D/dalvikvm( 9159): VFY: dead code 0x0009-0009 in Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.getNumColumnsCompatible ()I
02-22 07:49:04.963 I/dalvikvm( 9159): Could not find method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.smoothScrollToPositionFromTop, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.tryToScrollToBottomSmoothly
02-22 07:49:04.963 W/dalvikvm( 9159): VFY: unable to resolve virtual method 20600: Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.smoothScrollToPositionFromTop (II)V
02-22 07:49:04.963 D/dalvikvm( 9159): VFY: replacing opcode 0x6e at 0x0011
02-22 07:49:04.963 I/dalvikvm( 9159): Could not find method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.smoothScrollToPositionFromTop, referenced from method third_parties.in.srain.cube.GridViewWithHeaderAndFooter.tryToScrollToBottomSmoothly
02-22 07:49:04.963 W/dalvikvm( 9159): VFY: unable to resolve virtual method 20601: Lthird_parties/in/srain/cube/GridViewWithHeaderAndFooter;.smoothScrollToPositionFromTop (III)V
02-22 07:49:04.963 D/dalvikvm( 9159): VFY: replacing opcode 0x6e at 0x0011
02-22 07:49:04.973 W/dalvikvm( 9159): PGC: map but no entry for Landroid/view/LayoutInflater;.inflate addr=0xfff8e76a
02-22 07:49:05.043 D/dalvikvm( 9159): GC_EXTERNAL_ALLOC freed 60K, 20% free 7527K/9351K, external 528K/532K, paused 79ms
02-22 07:49:05.058 E/OCFileListFragment( 9159): onActivityCreated() start
02-22 07:49:05.063 I/dalvikvm( 9159): Could not find method android.widget.AbsListView.getChoiceMode, referenced from method com.owncloud.android.ui.adapter.FileListListAdapter.getView
02-22 07:49:05.068 W/dalvikvm( 9159): VFY: unable to resolve virtual method 8518: Landroid/widget/AbsListView;.getChoiceMode ()I
02-22 07:49:05.068 D/dalvikvm( 9159): VFY: replacing opcode 0x74 at 0x0152
02-22 07:49:05.068 D/dalvikvm( 9159): VFY: dead code 0x0155-015e in Lcom/owncloud/android/ui/adapter/FileListListAdapter;.getView (ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
02-22 07:49:05.068 D/dalvikvm( 9159): VFY: dead code 0x0227-024a in Lcom/owncloud/android/ui/adapter/FileListListAdapter;.getView (ILandroid/view/View;Landroid/view/ViewGroup;)Landroid/view/View;
02-22 07:49:05.073 D/appproc ( 9159): Shutting down VM
02-22 07:49:05.073 W/dalvikvm( 9159): threadid=1: thread exiting with uncaught exception (group=0x40015568)
02-22 07:49:05.078 E/AndroidRuntime( 9159): FATAL EXCEPTION: main
02-22 07:49:05.078 E/AndroidRuntime( 9159): java.lang.NoSuchMethodError: android.widget.AbsListView.setAdapter
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.owncloud.android.ui.fragment.ExtendedListFragment.setListAdapter(ExtendedListFragment.java:84)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.owncloud.android.ui.fragment.OCFileListFragment.onActivityCreated(OCFileListFragment.java:137)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.Fragment.performActivityCreated(Fragment.java:1794)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:967)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.FragmentManagerImpl.moveToState(FragmentManager.java:1126)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.BackStackRecord.run(BackStackRecord.java:739)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.FragmentManagerImpl.execPendingActions(FragmentManager.java:1489)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.support.v4.app.FragmentActivity.onStart(FragmentActivity.java:548)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.owncloud.android.ui.activity.FileActivity.onStart(FileActivity.java:182)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.owncloud.android.ui.activity.FileDisplayActivity.onStart(FileDisplayActivity.java:220)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.Instrumentation.callActivityOnStart(Instrumentation.java:1129)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.Activity.performStart(Activity.java:3815)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:1637)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:1680)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.ActivityThread.access$1500(ActivityThread.java:117)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.ActivityThread$H.handleMessage(ActivityThread.java:931)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.os.Handler.dispatchMessage(Handler.java:99)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.os.Looper.loop(Looper.java:130)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at android.app.ActivityThread.main(ActivityThread.java:3703)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at java.lang.reflect.Method.invokeNative(Native Method)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at java.lang.reflect.Method.invoke(Method.java:507)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:841)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:599)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at de.robv.android.xposed.XposedBridge.main(XposedBridge.java:132)
02-22 07:49:05.078 E/AndroidRuntime( 9159): 	at dalvik.system.NativeStart.main(Native Method)
02-22 07:49:05.088 W/ActivityManager( 5060):   Force finishing activity com.owncloud.android/.ui.activity.FileDisplayActivity
~~~

**Comment by tobiasKaminsky** [comment]
I think the best workaround for still supporting old devices is to disable the grid view for them.

**Comment by smu** [comment]
I am using an HTC wildfire with Android 2.3 every day therefore:

+1 for owncloud on android 2.x

**Comment by flyingOwl** [comment]
The "breaking" commit is probably this one: 2cfed6eaa79df1e5b18eedc356576d66b0a74c63
It's the one that pops up if you search for the missing method "setAdapter", which btw. was introduced with Android 3.x (API level 11).

For everyone on Android 2.x, you can downgrade to v1.6.2 via F-Droid:
https://f-droid.org/repository/browse/?fdid=com.owncloud.android

**Comment by TtuxX** [comment]
Hello @davivel ! Could you possibly confirm whether or not ownCloud Android app is likely to crash on 2.x Android platform since commit https://github.com/owncloud/android/commit/2cfed6eaa79df1e5b18eedc356576d66b0a74c63 as pointed out by @flyingOwl ? Many thanks in advance :)

**Comment by masensio** [comment]
Hi all,

We are working to fix this bug. 
It will be fix in the next release of the app (1.7.1).  :relaxed: 
The new version will be released in a short time.

Thanks all for the feedback an patience.

**Comment by TtuxX** [comment]
Hi @masensio 
Thank you very much for all your time & hard work ! thank you also for supporting this time once again Android 2.x, that's really nice from you :)

**Comment by masensio** [comment]
This fix is into develop branch. :relaxed: 
Ready to be released in the next version of the app (1.7.1)
